docs/howto/virtual-host-addressing.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: default
+title: "S3 Virtual-host addressing (advanced)"
+description: Configure the lakeFS S3 API to use virtual host addressing
+parent: How-To
+---
+
+# Configuring lakeFS to use S3 Virtual-Host addressing
+
+## Understanding virtual-host addressing
+
+Some systems require S3 endpoints (such as lakeFS's [S3 Gateway](../understand/architecture.md#s3-gateway)) to support [virtual-host style addressing](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html){: target="_blank" }.
+
+lakeFS supports this, but requires some configuration in order to extract the bucket name (used as the lakeFS repository ID) from the host address.
+
+For example:
+
+```text
+GET http://foo.example.com/some/location
+```
+
+In this case, there's no way for lakeFS to determine whether this is a virtual-host request where the endpoint url is `example.com`, the bucket name is `foo` and the path is `/some/location`,
+or a path-based request where the endpoint is `foo.example.com`, the bucket name is `some` and the path is `location`.
+
+This requires an extra step: Defining an explicit set of DNS record for lakeFS S3 gateway.
```

Comment:
```suggestion
```suggestion
There are two ways to interpret the URL above: 
- as a virtual-host URL where the endpoint URL is `example.com`, the bucket name is `foo`, and the path is `/some/location`
- as a path-based URL where the endpoint is `foo.example.com`, the bucket name is `some` and the path is `location`

By default, lakeFS reads URLs as path-based. To read it a virtual-host request lakeFS requires additional configuration: defining an explicit set of DNS records for the lakeFS S3 gateway.
```

Suggesting to change the phrasing to mention what lakeFS can do rather than what it can't do, and mention what happens by default. Please correct me If I'm not accurate with what happens by default

Reply:
Fixed it a little